src/runtime_src/core/edge/drm/zocl/zocl_ioctl.c

```diff
@@ -230,60 +225,29 @@ zocl_fpga_mgr_load(struct drm_device *ddev, char *data, int size)
 	return err;
 }
 
-static int zocl_pcap_download(struct drm_zocl_dev *zdev,
-			      const void __user *bit_buf, unsigned long length)
+static int
+zocl_load_bitstream(struct drm_zocl_dev *zdev, char *buffer, int length)
 {
 	XHwIcap_Bit_Header bit_header;
-	struct fpga_image_info *info = NULL;
-	char *buffer = NULL;
 	char *data = NULL;
 	unsigned int i;
 	char temp;
 	int err;
 	void __iomem *map = NULL;
 
 	memset(&bit_header, 0, sizeof(bit_header));
-	buffer = kmalloc(BITFILE_BUFFER_SIZE, GFP_KERNEL);
-
-	if (!buffer) {
-		err = -ENOMEM;
-		goto free_buffers;
-	}
-
-	if (copy_from_user(buffer, bit_buf, BITFILE_BUFFER_SIZE)) {
-		err = -EFAULT;
-		goto free_buffers;
-	}
-
 	if (bitstream_parse_header(buffer, BITFILE_BUFFER_SIZE, &bit_header)) {
-		err = -EINVAL;
-		goto free_buffers;
+		DRM_ERROR("bitstream header parse failed");
+		return -EINVAL;
 	}
 
 	if ((bit_header.HeaderLength + bit_header.BitstreamLength) > length) {
-		err = -EINVAL;
-		goto free_buffers;
-	}
-
-	bit_buf += bit_header.HeaderLength;
-
-	data = vmalloc(bit_header.BitstreamLength);
-	if (!data) {
-		err = -ENOMEM;
-		goto free_buffers;
-	}
-
-	if (copy_from_user(data, bit_buf, bit_header.BitstreamLength)) {
-		err = -EFAULT;
-		goto free_buffers;
-	}
-
-	if (zdev->pr_isolation_addr) {
-		map = ioremap(zdev->pr_isolation_addr, 0x1000);
-		if (!IS_ERR_OR_NULL(map))
-			iowrite32(0x0, map);
+		DRM_ERROR("bitstream header+stream length parse failed");
+		return -EINVAL;
 	}
 
+	/* Byte swap bitstream data */
+	data = buffer + bit_header.HeaderLength;
 	for (i = 0; i < bit_header.BitstreamLength ; i = i+4) {
```

Comment:
it is OK for now. But I think it will be more efficient we use what has been provided by byteorder.h, e.t. le32_to_cpu(), etc.

Reply:
> it is OK for now. But I think it will be more efficient we use what has been provided by byteorder.h, e.t. le32_to_cpu(), etc.

add comments here